mysecondwebapp/src/main/resources/META-INF/resources/WEB-INF/jsp/material.jsp

```diff
@@ -54,12 +55,17 @@
                 <div class="col-md-4 form-group">
                     <label for="category">Category</label>
                     <select name="category" id="category" class="form-control">
-                        <option value="0">Select category</option>
-                        <option value="1">Category1</option>
-                        <option value="2">Category2</option>
+                    	<option value="0">Select category</option>
+                    	<c:forEach items = "${materialCategory}" var = "materialCategory">
```

Comment:
Phía backend đang chỉ xử lý khi TH 1 material nằm trong 1 category.
Nếu FE chọn nhiều hơn 1 category thì code của backend có vẫn chạy đúng được ko?

Reply:
Ý chị ở đây là phần search hả, kiểu search name cùng với nhiều hơn 1 category hả, đoạn này cu chưa hiểu chị đang đề cập đến phần nào

Reply:
Thế cái file jps này dùng cho phần nào?
Thao tác thử đi rồi biết. add/update 1 cái material nằm trong nhiều hơn 1 category

Reply:
File này là để cu xử lý phần search và hiển thị table
Còn add/update 1 cái material nằm trong nhiều hơn 1 category thì cu đã xử lý theo cách như sau:
1.Phần Add thì cu đã tạo 1 file addMaterial.jsp nhận dữ liệu từ form thì phần category cu đã sử dụng checkbox để có thể chọn nhiều hơn 1 category rồi gửi nó qua backend để xử lý 
2. Phần update thì cu cũng đã tạo 1 file uodateMaterial.jsp nhưng phần category thì cu đã thay đổi chỉ cho chọn 1 category và khi truyền dữ liệu qua backend sẽ có trường hợp đã tồn tại material với category tương ứng thì sẽ chuyển hướng lại sang file update và hiển thị một thông báo lỗi "The material with the given category already exists."

Reply:
Tại sao update lại chỉ được thay đổi 1 category?

Reply:
1 material có thể nằm trong nhiều hơn 1 category nhưng ở table thì mỗi hàng 1 material sẽ ứng với 1 category nên cu sẽ update material ứng với category ấy

Reply:
Nghĩ xem làm thế nào để update được nhiều category đi